### PR TITLE
Rewrite remaining java in clojure

### PR DESCRIPTION
**Pull request by rufoa**
- slf4j Markers don't yet do anything special
  - eventual aim is to integrate them with timbre properly so people can filter by them
- no MDC anything yet
- todo: test for logger getName

### CODE REVIEW

**Comment by rufoa** [comment]
@fzakaria

I think this is basically ready for an initial release if you think it's okay? Remaining features can be added later

File: project.clj [hunk 1]
```diff
@@ -1,25 +1,18 @@
-(defproject com.fzakaria/slf4j-timbre "0.2.3"
+(defproject com.fzakaria/slf4j-timbre "0.3.0-SNAPSHOT"
```
Comment (fzakaria) [inline project.clj]: we can probably go straight to 0.3 since usage base is small (avoid snapshot)

**Comment by rufoa** [comment]
agreed, done

I'll let you merge + push to clojars when you're satisfied

**Comment by fzakaria** [comment]
Looks good.

I'm using Golang for a current project so I hadn't had a chance to give it a whirl.

I did notice though that Timbre has a `with-context` variable for the logging.
Seems like a good fit for MDC if its ever neded.

**Comment by fzakaria** [comment]
heading to gym. will merge & publish tonight.

**Comment by fzakaria** [comment]
LGTM

**Comment by rufoa** [comment]
> heading to gym. will merge & publish tonight.

cool

> I did notice though that Timbre has a with-context variable for the logging.
> Seems like a good fit for MDC if its ever neded.

yeah this was the plan, probably in conjunction with some custom timbre middleware

**Comment by fzakaria** [comment]
@rufoa  I think there's a bug with the branch.

`[org.slf4j/slf4j-api "1.7.14"]` needs to be listed as a full dependency not just in profiles

**Comment by rufoa** [comment]
good catch, will fix in a sec

**Comment by fzakaria** [comment]
Do we still need the reader dependency too?

**Comment by rufoa** [comment]
not sure. it was introduced in https://github.com/fzakaria/slf4j-timbre/commit/2a00c978e630edca9aeb6cae95844341c409ddfc . will investigate

**Comment by fzakaria** [comment]
so it was introduced when Timbre was using an older version (or pulled in a older one).
I think since we updated it, we no longer need to direct declare it
(my understanding of the AOT)

**Comment by rufoa** [comment]
makes sense and tests pass without it. I've pushed the changes to master

**Pull request merged**